backend/src/openbeheer/utils/open_zaak_helper/data_creation.py

```diff
@@ -0,0 +1,180 @@
+from dataclasses import dataclass
+from typing import Type
+from uuid import UUID
+from msgspec import UNSET, UnsetType
+from msgspec.json import decode
+from openbeheer.clients import ztc_client
+from openbeheer.types.ztc import (
+    Catalogus,
+    InformatieObjectType,
+    ResultaatType,
+    RolType,
+    ZaakType,
+    ZaakTypeInformatieObjectType,
+)
+from random import choice
+import string
+
+
+@dataclass
+class OpenZaakDataCreationHelper:
+    service_identifier: str
+    catalogus_uuid: UUID | UnsetType = UNSET
+
+    @property
+    def catalogus_url(self) -> str:
+        if not self.catalogus_uuid:
+            return ""
+
+        with ztc_client(self.service_identifier) as client:
+            return f"{client.base_url}catalogussen/{self.catalogus_uuid}"
+
+    def set_catalogus_uuid(self, catalogus_uuid: UUID) -> None:
+        self.catalogus_uuid = catalogus_uuid
```

Comment:
Why the setter?
I might even go as far as to make it a frozen dataclass; to prevent test isolation foot guns.